### PR TITLE
Buffs arm implants

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

You can now "drop" an item from an arm implant and it will automatically snap back into your arm and re-open the radial menu. This makes it much easier to do surgery, or use a tool implant. When the menu opens and you actually want to close it and not pick a new tool, you can click the red X in the center. NODROP still applies so these items cannot be shoved off of you, only you can drop them.

## Why It's Good For The Game
Arm implants are really awkward to use, this makes them much more usable.

## Images of changes

dreamseeker_vWKM8KWiCX

## Testing

Tested tool implant, surgery implant, and CC combat implant. Switching between all the tools worked, along with having two arm implants. An implant only having one item works as it did previously, with no radial showing up, but you are now able to drop it.

## Changelog
:cl:
add: Dropping an item from an arm implant will now open the menu to select another tool.
/:cl:

### CODE REVIEW

**Comment by Contrabang** [comment]
Also I just want to say this PR will probably need extensive reviewing... signals and adding a var on item level. Which is VERY useful, and makes things a lot less worse, and hopefully shouldnt have any major effects on performance.

**Comment by Sirryan2002** [comment]
Why not just use `COMSIG_ITEM_DROPPED` instead of adding an entirely new system to the base `item` type. Instead of having new bitflags, signals, and adding more checks/var to the drop proc stack, you could keep all of this code limited to only the implant.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -131,6 +131,9 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 	/// Holder var for the item outline filter, null when no outline filter on the item.
 	var/outline_filter
 
+	/// A linked implant to teleport to
+	var/obj/item/organ/internal/cyberimp/linked_implant
```
Comment (hal9000PR) [inline code/game/objects/items.dm:135]: I am not sure adding a var on all items for implants is a good idea

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -815,3 +820,9 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /// Called on cyborg items that need special charging behavior. Override as needed for specific items.
 /obj/item/proc/cyborg_recharge(coeff = 1, emagged = FALSE)
 	return
+
+/obj/item/proc/retract_to_linked_implant()
+	SIGNAL_HANDLER_DOES_SLEEP
+	if(istype(linked_implant, /obj/item/organ/internal/cyberimp/arm))
+		var/obj/item/organ/internal/cyberimp/arm/arm_implant = linked_implant
+		arm_implant.retract_and_show_radial()
```
Comment (hal9000PR) [inline code/game/objects/items.dm:828]:
```
/// Signifies that this proc is used to handle signals, but also sleeps.
/// Do not use this for new work.
#define SIGNAL_HANDLER_DOES_SLEEP

```
try wrapping it in an INVOKE_ASYNC instead.

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 1 above)
Comment (Contrabang) [reply code/game/objects/items.dm]: removed

File: code/game/objects/items.dm (hunk 2 above)
Comment (Contrabang) [reply code/game/objects/items.dm]: invoke_async added

**Comment by Contrabang** [comment]
PR has been reworked to use mob signals instead of item signals

File: code/game/objects/items/devices/flash.dm [hunk 3]
```diff
@@ -248,15 +248,15 @@
...
 	desc = "A high-powered photon projector implant normally used for lighting purposes, but also doubles as a flashbulb weapon. Self-repair protocols fix the flashbulb if it ever burns out."
 	cooldown_duration = 2 SECONDS
-	var/obj/item/organ/internal/cyberimp/arm/flash/I = null
+	var/obj/item/organ/internal/cyberimp/arm/implant = null
 
 /obj/item/flash/armimplant/burn_out()
-	if(I && I.owner)
-		to_chat(I.owner, "<span class='warning'>Your [name] implant overheats and deactivates!</span>")
-		I.Retract()
+	if(implant && implant.owner)
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/flash.dm:254]:
```suggestion
	if(implant?.owner)
```

File: code/modules/mob/mob_defines.dm [hunk 4]
```diff
@@ -44,7 +44,7 @@
 	var/memory = ""
 	var/next_move = null
 	var/notransform = FALSE	//Carbon
-	var/hand = null
+	var/hand = null // True for left hand active, otherwise for right hand active
```
Comment (Sirryan2002) [inline code/modules/mob/mob_defines.dm:47]:
```suggestion
	/// True for left hand active, otherwise for right hand active
	var/hand = null 
```

**Review by Sirryan2002 — COMMENTED** [review]
two things, otherwise this is looking good code wise

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by S34NW — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]
Not the perfect solution but the alternative is a complete drop/unequip chain refactor. This will do for now.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**